File: C++/Easy/WrongAnswer/distribute-candies-to-people_3.cpp
class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people) {
        
        vector<int> ans(num_people, 0);
        int i = 0;
        while (candies > 0) {
            ans[i] += min(candies, i + 1);
            candies -= i + 1;
            i = (i + 1) % num_people;
        }
        return ans;
    }
};

File: Rust/Easy/CompileError/distribute-candies-to-people_3.rs
impl Solution {
    pub fn distribute_candies(candies: i32, num_people: i32) -> Vec<i32> {
        
        let mut ans = vec![0; num_people as usize];
        let mut i = 0;
        let mut candy = 1;
        while candies > 0 {
            ans[i] += candy;
            candies -= candy;
            candy += 1;
            i += 1;
            if i == ans.len() {
                i = 0;
            }
            if candies < candy {
                candy = candies;
            }
        }
        ans
    }
}